### PR TITLE
Add missing dict conversion in backconfiguration class

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

After #4016 has merged a missed edge case in the handling of to_dict()
and from_dict() for PulseBackendConfiguration was discovered via the aer
unit tests. The u_channel_lo attribute is a nested object and needs to
call UChannelLO's to_dict() and from_dict() methods in those respective
calls. However, this conversion was not being done. This commit corrects
the oversight.

### Details and comments

### CODE REVIEW

File: qiskit/providers/models/backendconfiguration.py [hunk 1]
```diff
@@ -486,9 +486,15 @@ def from_dict(cls, data):
...
-        gates = [GateConfig.from_dict(x) for x in data.pop('gates')]
-        data['gates'] = gates
-        return cls(**data)
+        in_data = copy.copy(data)
+        gates = [GateConfig.from_dict(x) for x in in_data.pop('gates')]
+        in_data['gates'] = gates
+        input_uchannels = in_data.pop('u_channel_lo')
+        u_channels = []
+        for channel in input_uchannels:
+            u_channels.append([UchannelLO.from_dict(x) for x in channel])
```
Comment (jyu00) [inline qiskit/providers/models/backendconfiguration.py:495]: So the type hint at line 363 is wrong. `u_channel_lo` should be `List[List[UchannelLO]]`.

**Comment by chriseclectic** [comment]
This still doesn't quite fix the Aer tests. Running tests on this branch there is still one test failure with
```
======================================================================
ERROR: test_control_channel_labels_from_backend (test.terra.openpulse.test_system_models.TestPulseSystemModel)
Test correct importing of backend control channel description.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "qiskit-aer/test/terra/openpulse/test_system_models.py", line 106, in test_control_channel_labels_from_backend
    system_model = PulseSystemModel.from_backend(backend)
  File "qiskit-aer/qiskit/providers/aer/pulse/pulse_system_model.py", line 159, in from_backend
    u_string += str(scale[0] + scale[1] * 1j) + 'q' + str(q_idx)
TypeError: 'complex' object is not subscriptable
```

I'm guessing this is to do with JSON storing complex as a pair vs using complex in python dict?

**Comment by mtreinish** [comment]
@chriseclectic yeah, that looks like the pulse system model was assuming `[real, imag]`. That should be fixed by: https://github.com/Qiskit/qiskit-aer/pull/694

File: qiskit/providers/models/backendconfiguration.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/providers/models/backendconfiguration.py:495]: Fixed in: https://github.com/Qiskit/qiskit-terra/pull/4124/commits/14ca7e107e8fa0da9a72aa03b30353c41322b2de
